src/main/java/co/rsk/federate/FederationProviderFromFederatorSupport.java

```diff
@@ -156,31 +150,26 @@ private Federation getExpectedFederation(Federation initialFederation, Address e
         if (initialFederation.getAddress().equals(expectedFederationAddress)) {
             return initialFederation;
         }
-
-        // If addresses do not match build an ERP federation
-        Federation erpFederation = FederationFactory.buildNonStandardErpFederation(
-            initialFederation.getMembers(),
-            initialFederation.getCreationTime(),
-            initialFederation.getCreationBlockNumber(),
-            federatorSupport.getBtcParams(),
-            bridgeConstants.getErpFedPubKeysList(),
-            bridgeConstants.getErpFedActivationDelay(),
-            federatorSupport.getConfigForBestBlock()
-        );
-
-        if (erpFederation.getAddress().equals(expectedFederationAddress)) {
-            return erpFederation;
+        List<FederationMember> members = initialFederation.getMembers();
+        Instant creationTime = initialFederation.getCreationTime();
+        long creationBlockNumber = initialFederation.getCreationBlockNumber();
+        NetworkParameters btcParams = federatorSupport.getBtcParams();
+        List<BtcECKey> erpPubKeys = bridgeConstants.getErpFedPubKeysList();
+        long activationDelay = bridgeConstants.getErpFedActivationDelay();
+        ActivationConfig.ForBlock activations = federatorSupport.getConfigForBestBlock();
+
+        ErpFederationArgs erpFederationArgs =
+            new ErpFederationArgs(members, creationTime, creationBlockNumber, btcParams, erpPubKeys, activationDelay);
+
+        // If addresses match build a Non-Standard ERP federation
```

Comment:
Following the clean code advice we have seen recently, I don't think we need these types of comments. The code below is easy to read and says clearly what it does:

```java
if (nonStandardErpFederation.getAddress().equals(expectedFederationAddress)) {
            return nonStandardErpFederation;
}
```

I think this is clear enough, not need for a documentation comment in this case.